### PR TITLE
PHP 7.0 deprecates salt option. Let's disallow it too.

### PR DESCRIPTION
**Pull request by QWp6t**
> **Warning** The salt option has been deprecated as of PHP 7.0.0. It is now preferred to simply use the salt that is generated by default.

http://php.net/manual/en/function.password-hash.php

### CODE REVIEW

**Comment by swalkinshaw** [comment]
I don't think it's up to us to enforce deprecations in PHP core functions. We're simply allowing people to set options via a filter. It's up to them to know the options and deprecations.

**Comment by QWp6t** [comment]
I think you might be misinterpreting my intention here. I didn't create this PR because I actually care about enforcing PHP's  deprecations. It's about removing the option *for the same reason that it was deprecated* in PHP 7: it ensures the passwords are secure.

There are two reasons why this PR is a good idea.

1. Random salts are more secure. There is no reason to use a predictable value (static or otherwise), and since PHP already generates random salts on its own if this value isn't set, there is no reason to set the value at all.

2. This PR hard codes a default `cost` option to allow us to backport increased cost to previous versions of PHP as the default value gets increased in future versions. e.g., if PHP 7.1 changes the default `cost` to 15, then we can make that change here and it would become the default regardless of what version of PHP is running on the server.

In short, this PR ensures that developers do not inadvertently make their password hashes less secure.

Given that there is only the potential for a benefit from this PR and there are no downsides to it, I find it both odd and troubling that it would be rejected. I won't press the issue any further than this. Feel free to close it if you're still not in favor of it.

Kind Regards,
QW6t

PS-
To assist you with your decision, I have created an exhaustive list of pros and cons.

**Pros**
 - More secure salts ensures more secure hashes
 - Up to date default `cost` ensures more secure hashes

**Cons**
 - None.

**Comment by Foxaii** [comment]
There is a con. By using a non-system default, this PR would require users to update the plugin whenever a newer version of PHP increases the cost. That's not a lot of maintenance to ask of a developer, but could easily be missed by end users, more so given that we suggest the plugin is used as a composer dependancy or mu-plugin.

**Comment by renatonascalves** [comment]
I'm using PHP 7+, does that mean my salt hash is less secure? This PR seems to imply that. :/

**Comment by swalkinshaw** [comment]
@renatonascalves no, not at all. This PR just makes it so that a user can't manually specify a salt option which is deprecated. But our plugin doesn't set a manual salt anyway.

**Comment by swalkinshaw** [comment]
Closing this as is. Looks like PHP 7.2 might get Argon2 which takes 3 options. This PR would make it even more incompatible.
